File: typescript/easy/divide-array-into-equal-pairs.ts
// https://leetcode.com/problems/divide-array-into-equal-pairs/description/

function divideArray(nums: number[]): boolean {
  const pairs = {};

  for (const num of nums) {
    pairs[num] = pairs[num] + 1 || 1;
  }

  for (const key in pairs) {
    if (pairs[key] % 2 !== 0) {
      return false;
    }
  }
  return true;
}

File: go/divide-array-into-equal-pairs.go
// https://leetcode.com/problems/divide-array-into-equal-pairs/description/

package main

func divideArray(nums []int) bool {
	pairs := make(map[int]int)

	for _, num := range nums {
		pairs[num]++
	}

	for _, v := range pairs {
		if v%2 != 0 {
			return false
		}
	}

	return true
}
